pytest_fixtures/satellite_auth.py

```diff
@@ -248,48 +251,70 @@ def enroll_configure_rhsso_external_auth():
 
 
 @pytest.fixture(scope='session')
-def enable_external_auth_rhsso(enroll_configure_rhsso_external_auth):
+def enable_external_auth_rhsso(enroll_configure_rhsso_external_auth, default_sat):
     """register the satellite with RH-SSO Server for single sign-on"""
     client_id = get_rhsso_client_id()
     create_mapper(GROUP_MEMBERSHIP_MAPPER, client_id)
     audience_mapper = copy.deepcopy(AUDIENCE_MAPPER)
     audience_mapper['config']['included.client.audience'] = audience_mapper['config'][
         'included.client.audience'
-    ].format(rhsso_host=settings.server.hostname)
+    ].format(rhsso_host=default_sat)
     create_mapper(audience_mapper, client_id)
     set_the_redirect_uri()
 
 
 @pytest.mark.external_auth
 @pytest.fixture(scope='session')
-def enroll_idm_and_configure_external_auth():
+def enroll_idm_and_configure_external_auth(default_sat):
     """Enroll the Satellite6 Server to an IDM Server."""
     run_command(cmd='yum -y --disableplugin=foreman-protector install ipa-client ipa-admintools')
-
     run_command(
         cmd=f'echo {settings.ipa.password} | kinit admin',
         hostname=settings.ipa.hostname,
     )
-    result = run_command(
-        cmd=f'ipa host-add --random {settings.server.hostname}',
-        hostname=settings.ipa.hostname,
-    )
+    output = ssh.command(f'ipa host-find {default_sat.hostname}')
```

Comment:
For these cases, since the command is being run on the `default_sat` instance, you should use `default_sat.execute()` here instead of `ssh.command`. `ssh.command` is still relying on trying to read settings.server.hostname.

Reply:
Its a mix here of commands against `settings.ipa.hostname` and default_sat, this comment applies to all the places where you're running commands against default_sat.  Using `ssh.command` for the commands send to the IPA host is acceptable - though not ideal.  Ideal would be creating a host instance for it.

```
from robottelo.hosts import ContentHost

ipa_host = ContentHost(settings.ipa.hostname)

default_sat.execute('yum -y --disableplugin=foreman-protector install ipa-client ipa-admintools')
ipa_host.execute(f'echo {settings.ipa.password} | kinit admin')
output = default_sat.execute(f'ipa host-find {default_sat.hostname}')
if output.status != 0:
    result = ipa_host.execute(f'ipa host-add --random {default_sat.hostname}')
    <etc>
```